### DISCUSSION TITLE
High-speed, adaptive streaming video player for web

### DISCUSSION THREAD

**Discussion by ahmetilhn** [community/community#129416] [Programming Help]
### Select Topic Area

Product Feedback

### Body

Hello guys, I'm working on a future-oriented video player. If anyone wants to give advice or support on this project, they can contact me. Thanks 💀

![Mobile](https://github.com/community/community/assets/60369989/44761d57-fa39-492c-9647-8e8192da9256)

Repository: https://github.com/ahmetilhn/savior-video-player
Mail: [ahmetilhan.dev@gmail.com](mailto:ahmetilhan.dev@gmail.com)

**Comment by ahmetilhn** [comment]
It is necessary to make such a package with React and distribute it to all applications. We will be able to use this package in Vue, react, svelte, jquery and plain html projects. The reason for such a wide distribution area is that the majority of movie sites work with jquery or plain HTML.

**Comment by ahmetilhn** [comment]
A user enters a site to watch a video. The length of the video is 24 seconds. But since this man's internet connection is low, he can only receive a total of 12 seconds of video data in 24 seconds. Now, we cannot increase this man's internet connection with the available resources, but with an approach like this, we can reduce the man's total freezing time. The man opened the video and started it and reached the 8th second (it still didn't freeze). If we reduce the speed of the video from 1x to 0.8x starting from this second, the man will normally watch the 4 seconds between 8 and 12 as 5 seconds, instead of freezing at the 12th second. In this case, the man is caught in the 13th second instead of the 12th second. 1 second may seem small in this scenario, but let's assume that a movie is 120 * 60 on average, it is 7200 seconds in total. 7200 / 12 is 600. That's 10 minutes, so we'll have this guy freeze for 10 minutes less throughout the movie. And while we do this, we do it at 0.8x speed, slowing down so little that it is not noticeable.

**Comment by ebndev** [comment]
Hi @ahmetilhn, thanks for sharing your project with the community! 

Unfortunately, we currently do not allow self-promotion or solicitation in Community Discussions. We want to make sure there is space for users to ask questions without overwhelming them with other conversations. Please feel encouraged to post if you have specific questions  or need programming help while you're working through it.
